internal/controller/storageclient_controller.go

```diff
@@ -239,6 +241,23 @@ func (r *storageClientReconcile) reconcile(ctx context.Context, req ctrl.Request
 
 	result, reconcileErr := r.reconcilePhases()
 
+	storageClaimList := &metav1.PartialObjectMetadataList{}
+	storageClaimList.SetGroupVersionKind(v1alpha1.GroupVersion.WithKind("StorageClaimList"))
+	if err := r.list(storageClaimList); err != nil {
+		r.log.Info("No StorageClaims remainings")
+	}
+
+	for _, storageClaim := range storageClaimList.Items {
+		if err := r.Client.Delete(r.ctx, &storageClaim); client.IgnoreNotFound(err) != nil {
```

Comment:
you are iterating on a list of objects and no need to ignore the not found error, if it indeed gets deleted b/n the list and delete let it requeue w/ an error and it'll be removed eventually.